### PR TITLE
Use the chocolatey CDN directly to avoid the flaky API

### PR DESCRIPTION
**Pull request by aidanhs**

### CODE REVIEW

File: src/ci/scripts/install-msys2.sh [hunk 1]
```diff
@@ -12,10 +12,12 @@ IFS=$'\n\t'
...
 
 if isWindows; then
-    for RETRY_COUNT in 1 2 3 4 5 6 7 8 9 10; do
-        choco install msys2 \
-            --params="/InstallDir:$(ciCheckoutPath)/msys2 /NoPath" -y --no-progress \
-            && mkdir -p "$(ciCheckoutPath)/msys2/home/${USERNAME}" \
-            && ciCommandAddPath "$(ciCheckoutPath)/msys2/usr/bin" && break
-    done
+    # Pre-followed the api/v2 URL to the CDN since the API can be a bit flakey
+    curl -sSL https://packages.chocolatey.org/msys2.20190524.0.0.20191030.nupkg > msys2.nupkg
```
Comment (Mark-Simulacrum) [inline src/ci/scripts/install-msys2.sh]: Have we confirmed that the *API* and not the package registry is flaky? If we're going to pin to a specific version, maybe we should upload these files to S3 and be done with it?

Comment (mati865) [reply src/ci/scripts/install-msys2.sh]: > Have we confirmed that the API and not the package endpoint is flaky?

Sort of: https://status.chocolatey.org/issues/2019-12-18-timeouts-when-searching-and-installing-packages/

Comment (aidanhs) [reply src/ci/scripts/install-msys2.sh]: The errors I've seen directly point at the API (https://github.com/rust-lang/rust/issues/67568) and the headers from the CDN indicate it's backed by cloudfront which I figure should be reliable.

This makes it easier to bump versions as well.

Comment (Mark-Simulacrum) [reply src/ci/scripts/install-msys2.sh]: Okay, great! That sounds like a good reason to prefer the packages endpoint for now, and we can always take the next step if it becomes necessary.

**Comment by Mark-Simulacrum** [comment]
r=me with tidy fixed.

**Comment by aidanhs** [comment]
Just want to note one test scenario I did try out successfully: downloading the nupkgs, disconnecting from internet and doing the install, to make sure it wasn't trying to do any other API operations.

File: src/ci/scripts/install-msys2.sh [hunk 2]
```diff
@@ -12,10 +12,14 @@ IFS=$'\n\t'
...
 
 if isWindows; then
-    for RETRY_COUNT in 1 2 3 4 5 6 7 8 9 10; do
-        choco install msys2 \
-            --params="/InstallDir:$(ciCheckoutPath)/msys2 /NoPath" -y --no-progress \
-            && mkdir -p "$(ciCheckoutPath)/msys2/home/${USERNAME}" \
-            && ciCommandAddPath "$(ciCheckoutPath)/msys2/usr/bin" && break
-    done
+    # Pre-followed the api/v2 URL to the CDN since the API can be a bit flakey
+    curl -sSL https://packages.chocolatey.org/msys2.20190524.0.0.20191030.nupkg > \
```
Comment (lzutao) [inline src/ci/scripts/install-msys2.sh]:
```suggestion
    curl -sSL https://packages.chocolatey.org/msys2.20190524.0.0.20191030.nupkg -o msys2.nupkg
```

**Pull request merged**